### PR TITLE
Request and display query results for multiplecategories

### PR DESCRIPTION
**Pull request by mantagen**
- results now sent be server as
```
[{
  country: countryId,
  category: categoryId,
  result: queryResult 
}, ...]
```

- results stored in elm as
```
DictList (CategoryId, CountryId) QueryResult
```

### CODE REVIEW

File: server/src/router/query.route.js [hunk 1]
```diff
@@ -4,20 +4,39 @@ export default ({ config, searchApiService }) => async (req, res, next) => {
...
+            countries: [country],
+            categories: [category],
+            filterText
+          })
+        };
+      });
+
+    const queries = categories
+      .map(categoryRequest)
+      .reduce((accum, curr) => accum.concat(curr), []);
```
Comment (Jbarget) [inline server/src/router/query.route.js]: not sure why you need this `reduce`? 

Is the result of
```
    const queries = categories
      .map(categoryRequest)
```

not an array of objects?

File: server/src/router/query.route.js [hunk 2]
```diff
@@ -4,20 +4,39 @@ export default ({ config, searchApiService }) => async (req, res, next) => {
...
+          ...query,
+          result: result.data
+        };
       })
     );
 
-    const resultsRes = await Promise.all(requests);
-
-    const results = resultsRes.map(result => result.data);
+    console.log('sdfds', responses[0]);
```
[the remark this answers is not in the document]
Comment (mantagen) [reply server/src/router/query.route.js]: hehehehe

File: server/src/router/query.route.js (hunk 1 above)
Comment (mantagen) [reply server/src/router/query.route.js]: The reduce part flattens it from 
`[[{}, {}], [{}, {}]]`
to
`[{}, {}, {}, {}]`

File: client/src/Update/Update.elm [hunk 3]
```diff
@@ -445,46 +446,50 @@ update msg model =
...
-                        0 ->
-                            countrySelect
-                                |> Dict.insert 0 { countrySelect1 | query = countrySelect2.query }
-                                |> Dict.insert 1 { countrySelect2 | query = "" }
-
-                        1 ->
-                            countrySelect
-                                |> Dict.insert 1 { countrySelect2 | query = "" }
+                countryId =
+                    second categoryCountry
```
[the remark this answers is not in the document]
Comment (mantagen) [reply client/src/Update/Update.elm]: Agree with first option

**Pull request merged**